hedera-bls-impl/src/main/java/com/hedera/platform/bls/impl/LibraryLoader.java

```diff
@@ -0,0 +1,363 @@
+        if (isJarFile(fullJarPathURL)) {
+            File extracted =
+                    nestedExtract(tempDirectory, fullJarPathURL.resolve(relativePath).toString());
+
+            if (extracted != null) {
+                return extracted;
+            }
+        }
+
+        // Try to get the file/directory straight from the file system
+        return getFileFromFileSystem(relativePath, tempDirectory);
+    }
+
+    /**
+     * Does the path lead to a valid JAR file?
+     *
+     * @param jarPath the path of the potential jar
+     * @return true if the provided path is a jar file, otherwise false
+     */
+    private static boolean isJarFile(final Path jarPath) {
+        if (jarPath == null) {
+            return false;
+        }
+
+        try (JarFile jarFile = new JarFile(jarPath.toString())) {
+            // Successfully opened the jar file. Check if there's a manifest
+            return jarFile.getManifest() != null;
+        } catch (final IOException | IllegalStateException | SecurityException e) {
+            logger.debug("This is not a JAR file due to {}", e.getMessage());
+
+            return false;
+        }
+    }
+
+    /**
+     * If we're not in a JAR, then we can load directly from the file system without all the
+     * unzipping fiasco present in {@see #getFileFromJar}.
+     *
+     * @param relativePath A relative path to a file or directory in the resources folder.
+     * @param outputDirectory A directory in which to store loaded files. Preferentially a temporary
+     *     one.
+     * @return The file or directory that was requested.
+     * @throws IOException Could not find requested file.
+     */
+    private static File getFileFromFileSystem(final Path relativePath, final File outputDirectory)
+            throws IOException, URISyntaxException {
+
+        final URL url =
+                ResourceLoader.class.getResource(
+                        prefixStringWithSlashIfNotAlready(relativePath.toString()));
+
+        if (url == null) {
+            throw new IOException("unable to get resource");
+        }
+
+        final File file;
+        if (Platform.isWindows()) {
+            file = Paths.get(url.toURI()).toFile();
+        } else {
+            file = new File(url.getFile());
+        }
+
+        if (!file.isFile()) {
+            throw new IOException("resource to load should be a file");
+        }
+
+        File resource = new File(relativePath.toUri());
+        File resourceCopiedToTempFolder = new File(outputDirectory, resource.getName());
+        FileUtils.copyFile(file, resourceCopiedToTempFolder);
+
+        return resourceCopiedToTempFolder;
+    }
+
+    /**
+     * If the string does not start with a slash, then make sure it does.
+     *
+     * @param s A string to prefix
+     * @return A string with a slash prefixed
+     */
+    private static String prefixStringWithSlashIfNotAlready(String s) {
+        if (!s.startsWith("/")) {
+            s = "/" + s;
+        }
```

Comment:
```suggestion
    private static String prefixStringWithSlashIfNotAlready(final String s) {
        if (!s.startsWith("/")) {
            s = "/" + s;
        }
```

This is the kind of stuff that is handled more elegantly by the path APIs.